src/dymoprint/lib/render_engines/margins.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+import math
+from enum import Enum
+
+from PIL import Image
+
+from dymoprint.lib.render_engines.render_context import RenderContext
+from dymoprint.lib.render_engines.render_engine import (
+    RenderEngine,
+    RenderEngineException,
+)
+
+
+class BitmapTooBigError(RenderEngineException):
+    def __init__(self, width_px, max_width_px):
+        msg = f"width_px: {width_px}, max_width_px: {max_width_px}"
+        super().__init__(msg)
+
+
+class MarginsMode(Enum):
+    PRINT = 1
+    PREVIEW = 2
```

Comment:
I prefer not to. I think it is more obvious to have a single place where the possible values are.
I think this way it is better than having to add `Literal` type hint to each argument in the function call chain, where in each such type hint we have to specify all accepted values. This is what `Enum`s are for, no?

Reply:
Yes, that's what `Enum`s are for, but I disagree that it's more obvious to have the possible values in a single place.

Probably the most important metric of code quality to me is, "How much do I have to poke around in the code before I understand what's going on?" When using an `Enum` as opposed to `Literal` it means that the given block of code no longer stands on its own. Whenever I want to understand or use it (e.g. from a REPL) I must look into another file to read the definition and then either import the `Enum` or make use of the meaningless correspondence between names and values.

The obvious counterpoint is `Enum`s are good because they're DRY, and DRY is good. But why is DRY good? It usually reduces code length, but I prioritize clarity over length. It's also good to prevent different parts of code from going out-of-sync. But with `Literal` you have the type checker for that. And the type checker will do a much better job with `Literal` than `Enum`. If you add a new member to your `Enum` you should update all functions that use it. With `Literal` the type checker should automatically flag all the functions that need to be updated. It's also more powerful because for example it's easy to define functions which only handle subsets of the keys.

`Enum`s aren't terrible, but unless you have a large number of elements or if there's a meaningful correspondence between names and values, I just don't see any benefit. My feeling is that they're mostly obsolete in modern Python. Ultimately it's a subjective style choice, so feel free to pass if you're unconvinced.

Reply:
I've commited a change to use literal type hint instead of enum. I feel less confident relying on the liner for detecting issues such as we had earlier during review, when in some places I've forgot to update the type literal. I personally favor DRY code (I really struggle not to fix wet code whenever I encounter it), which I believe improves code's maintainability. I think that prioritizing maintanability over readability (simplicity of the code) is the way to go. However, in this project, I do not want to enforce my personal preference, and I have no problem compromising to get to a consensus, which I believe would increase productivity and improve overall developer experience of all of us :-)